controllers/storage_test.go

```diff
@@ -181,3 +185,251 @@ func TestDefaultDeployBehaviorStorage(t *testing.T) {
+}
+
+func TestIsDatabaseAccessibleBadAccessKey(t *testing.T) {
+	// Override the live connection function with a mock version
+	ConnectAndQueryObjStore = func(ctx context.Context, log logr.Logger, endpoint, bucket string, accesskey, secretkey []byte, secure bool) bool {
+		return true
+	}
+
+	testNamespace := "testnamespace"
+	testDSPAName := "testdspa"
+
+	// Minimal Inputs
+	dspa := &dspav1alpha1.DataSciencePipelinesApplication{
+		Spec: dspav1alpha1.DSPASpec{
+			ObjectStorage: &dspav1alpha1.ObjectStorage{
+				DisableHealthCheck: false,
+			},
+		},
+	}
+	dspa.Name = testDSPAName
+	dspa.Namespace = testNamespace
+
+	// Create Context, Fake Controller and Params (unused)
+	ctx, _, reconciler := CreateNewTestObjects()
+
+	SecureConnection := false
+	params := &DSPAParams{
+		ObjectStorageConnection: ObjectStorageConnection{
+			Host:            "foo",
+			Port:            "1337",
+			Secure:          &SecureConnection,
+			AccessKeyID:     "this-is-not-a-base64-encoded-string",
+			SecretAccessKey: base64.StdEncoding.EncodeToString([]byte("foosecretkey")),
+		},
+	}
+
+	verified := reconciler.isObjectStorageAccessible(ctx, dspa, params)
+	assert.False(t, verified)
+}
+
+func TestIsDatabaseAccessibleBadSecretKey(t *testing.T) {
+	// Override the live connection function with a mock version
+	ConnectAndQueryObjStore = func(ctx context.Context, log logr.Logger, endpoint, bucket string, accesskey, secretkey []byte, secure bool) bool {
+		return true
+	}
+
+	testNamespace := "testnamespace"
+	testDSPAName := "testdspa"
+
+	// Minimal Inputs
+	dspa := &dspav1alpha1.DataSciencePipelinesApplication{
+		Spec: dspav1alpha1.DSPASpec{
+			ObjectStorage: &dspav1alpha1.ObjectStorage{
+				DisableHealthCheck: false,
+			},
+		},
+	}
+	dspa.Name = testDSPAName
+	dspa.Namespace = testNamespace
+
+	// Create Context, Fake Controller and Params (unused)
+	ctx, _, reconciler := CreateNewTestObjects()
+
+	SecureConnection := false
+	params := &DSPAParams{
+		ObjectStorageConnection: ObjectStorageConnection{
+			Host:            "foo",
+			Port:            "1337",
+			Secure:          &SecureConnection,
+			AccessKeyID:     base64.StdEncoding.EncodeToString([]byte("fooaccesskey")),
+			SecretAccessKey: "this-is-not-a-base64-encoded-string",
+		},
+	}
+
+	verified := reconciler.isObjectStorageAccessible(ctx, dspa, params)
+	assert.False(t, verified)
+}
+
+func TestJoinHostPort(t *testing.T) {
+	tests := map[string]struct {
+		host     string
+		port     string
+		expected string
+	}{
+		"host and port defined": {host: "somehost", port: "1234", expected: "somehost:1234"},
+		"empty port":            {host: "somehost", port: "", expected: "somehost"},
+		"empty host":            {host: "", port: "1234", expected: ":1234"}, // TODO: this obviously wouldn't actually work, should we handle this case?
+		"both empty":            {host: "", port: "", expected: ""},          // TODO: this obviously wouldn't actually work, should we handle this case?
```

Comment:
I think we should error if host is not provided,

Reply:
added a new commit to handle it, removed these TODO comments